src/tdb_encode_model.c

```diff
@@ -411,11 +412,13 @@ tdb_error make_grams(FILE *grouped,
     TDB_TIMER_END("encode_model/find_candidates")
 
     /* collect frequencies of *all* occurring bigrams of candidate unigrams */
-    TDB_TIMER_START
-    ret = event_fold(all_bigrams, grouped, num_events, items, num_fields, &g);
-    if (ret)
-        goto done;
-    TDB_TIMER_END("encode_model/all_bigrams")
+    if (!no_bigrams) {
```

Comment:
ok, if this PR makes a big impact as such, no reason to delay it further. We can add the further optimization in `choose_grams` later. Could you please add a comment in the `!no_bigrams` block ```
/* TODO: choose_grams below could be optimized for the nobigram case */
```